Simplified/MyBooks/NYPLMyBooksDownloadCenter.m

```diff
@@ -1620,5 +1638,56 @@ - (void)downloadProgressDidUpdateTo:(double)progress forBook:(NYPLBook * _Nonnul
 }
 #endif
 
+#if FEATURE_AUDIOBOOKS
+/// Create an audiobook object and add to download queue
+- (void)downloadAudiobookForBook:(NYPLBook *)book {
+  [AudiobookManifestAdapter transformAudiobookManifestWithBook:book
+                                                    completion:^(NSDictionary<NSString *,id> * _Nullable json,
+                                                                 id<DRMDecryptor> _Nullable decryptor,
+                                                                 enum AudiobookManifestError error) {
+    if (error == AudiobookManifestErrorNone) {
+      // Create audiobook
+      id<Audiobook> const audiobook = [AudiobookFactory audiobook:json decryptor:decryptor];
+      
+      if (!audiobook) {
+        NYPLLOG(@"Audiobook initiate failed");
+        // TODO: Handle Error
+        [self broadcastUpdate:book.identifier];
+        return;
+      }
+      AudiobookMetadata *metadata = [[AudiobookMetadata alloc] initWithTitle:book.title
+                                                                     authors:@[book.authors]];
+      
+      DefaultAudiobookManager *manager = [[DefaultAudiobookManager alloc] initWithMetadata:metadata
+                                                                                 audiobook:audiobook];
+      
+      [[NYPLBookRegistry sharedRegistry]
```

Comment:
it would be good to pass in the registry singleton instead of hardcoding it. Even better if it's a protocol.

Reply:
The `BookRegistry` singleton is widely used in this class, I am thinking of leaving it as-is now and we can fix it when we rewrite/refactor the `NYPLMyBooksDownloadCenter`. What do you think?

Reply:
yeah, that's true... now that you mention it, probably the bookRegistry should be an instance variable of the DownloadCenter.... but yes, that's a different story for a different ticket :)